testcontainers/azurite.py

```diff
@@ -0,0 +1,111 @@
+    _AZURITE_ACCOUNT_KEY = os.environ.get("AZURITE_ACCOUNT_KEY", "Eby8vdM02xNOcqFlqUwJPLlmEtlCDX"
+                                          "J1OUzFT50uSRZ6IFsuFq2UVErCz4I6"
+                                          "tq/K1SZFPTOtr/KBHBeksoGMGw==")
+
+    _BLOB_SERVICE_PORT = 10_000
+    _QUEUE_SERVICE_PORT = 10_001
+    _TABLE_SERVICE_PORT = 10_002
+
+    def __init__(
+            self,
+            image="mcr.microsoft.com/azure-storage/azurite:latest",
+            ports_to_expose=None,
+            **kwargs
+    ):
+        """ Constructs an AzuriteContainer.
+
+        Parameters
+        ----------
+        image: str
+            Expects an image with tag.
+        ports_to_expose: List[int]
+            Expects a list with port numbers to expose.
+        kwargs
+        """
+        super().__init__(image=image, **kwargs)
+
+        if ports_to_expose is None:
+            ports_to_expose = [
+                self._BLOB_SERVICE_PORT,
+                self._QUEUE_SERVICE_PORT,
+                self._TABLE_SERVICE_PORT
+            ]
+
+        if len(ports_to_expose) == 0:
+            raise ValueError("Expected a list with port numbers to expose")
+
+        self.ports_to_expose = ports_to_expose
+
+        self.with_exposed_ports(*ports_to_expose)
+        self.with_env("AZURITE_ACCOUNTS",
+                      f"{self._AZURITE_ACCOUNT_NAME}:{self._AZURITE_ACCOUNT_KEY}")
+
+    def get_connection_string(self):
+        host_ip = self.get_container_host_ip()
+        connection_string = f"DefaultEndpointsProtocol=http;" \
+                            f"AccountName={self._AZURITE_ACCOUNT_NAME};" \
+                            f"AccountKey={self._AZURITE_ACCOUNT_KEY};"
+
+        if self._BLOB_SERVICE_PORT in self.ports_to_expose:
+            connection_string += f"BlobEndpoint=http://{host_ip}:" \
+                                 f"{self.get_exposed_port(self._BLOB_SERVICE_PORT)}" \
+                                 f"/{self._AZURITE_ACCOUNT_NAME};"
+
+        if self._QUEUE_SERVICE_PORT in self.ports_to_expose:
+            connection_string += f"QueueEndpoint=http://{host_ip}:" \
+                                 f"{self.get_exposed_port(self._QUEUE_SERVICE_PORT)}" \
+                                 f"/{self._AZURITE_ACCOUNT_NAME};"
+
+        if self._TABLE_SERVICE_PORT in self.ports_to_expose:
+            connection_string += f"TableEndpoint=http://{host_ip}:" \
+                                 f"{self.get_exposed_port(self._TABLE_SERVICE_PORT)}" \
+                                 f"/{self._AZURITE_ACCOUNT_NAME};"
+
+        return connection_string
+
+    def start(self):
+        super().start()
+        self._connect()
+        return self
+
+    @wait_container_is_ready(OSError)
+    def _connect(self):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.connect((self.get_container_host_ip(),
+                       int(self.get_exposed_port(self.ports_to_expose[0]))))
```

Comment:
Does this always need to be `_BLOB_SERVICE_PORT` or could this be any port?

Reply:
This could be any port. If someone constructs just the queue service (like the example below) then it would be fine. 

```python
AzuriteContainer(ports_to_expose=[10_001])
```

The function is mainly to make sure if any connection can be made.